src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,247 @@
+
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+
+class NvmePoolCapacity(TestWithServers):
+    # pylfloat: disable=too-many-ancestors
+    """Test class Description: Verify NOSPC
+    condition is reported when accessing data beyond
+    pool size.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for test case."""
+        super(NvmePoolCapacity, self).setUp()
+
+        self.ior_flags = self.params.get("ior_flags", '/run/ior/iorflags/*')
+        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.ior_test_sequence = self.params.get("ior_test_sequence",
+                                                 '/run/ior/iorflags/*')
+        self.ior_daos_oclass = self.params.get("obj_class",
+                                               '/run/ior/iorflags/*')
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        self.pool = None
+        self.out_queue = queue.Queue()
+
+    def ior_thread(self, pool, oclass, api, test, flags, results):
+        """Start threads and wait until all threads are finished.
+        Args:
+            results (queue): queue for returning thread results
+
+        Returns:
+            None
+        """
+        processes = self.params.get("slots", "/run/ior/clientslots/*")
+        container_info = {}
+        mpio_util = MpioUtils()
+        if mpio_util.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test: Mpich not installed")
+        self.pool = pool
+        # Define the arguments for the ior_runner_thread method
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.set_daos_params(self.server_group, self.pool)
+        ior_cmd.daos_oclass.update(oclass)
+        ior_cmd.api.update(api)
+        ior_cmd.transfer_size.update(test[2])
+        ior_cmd.block_size.update(test[3])
+        ior_cmd.flags.update(flags)
+
+        container_info["{}{}{}"
+                       .format(oclass,
+                               api,
+                               test[2])] = str(uuid.uuid4())
+
+        # Define the job manager for the IOR command
+        manager = Mpirun(ior_cmd, mpitype="mpich")
+        manager.job.daos_cont.update(container_info
+                                     ["{}{}{}".format(oclass,
+                                                      api,
+                                                      test[2])])
+        env = ior_cmd.get_default_env(str(manager))
+        manager.setup_command(env, self.hostfile_clients,
+                              processes)
+
+        # run IOR Command
+        try:
+            manager.run()
+        except CommandFailure as _error:
+            results.put("FAIL")
+
+    def test_create_delete(self, num_pool=2, num_cont=5, total_count=100,
```

Comment:
How does this method test the create and delete? I don't see any value comparison or catching expected exception. display_pool_daos_space looks like just prints the info in the log.

Reply:
Yes. This is similar to metadata.py (test_metadata_addremove)... But, I am planning to insert an IOR operation in between and see whether IOR consumes correct space. I assume if there are memory leaks, eventually some thing will die... I will see whether I can implement some comparison method...